From 18edd9a58da066873dc2b231cd98c95cc30528ee Mon Sep 17 00:00:00 2001
From: Yong Zhi <yong.zhi@intel.com>
Date: Mon, 26 Feb 2018 15:34:06 -0600
Subject: [PATCH 014/129] media: intel-ipu4: isys: v4l2-compliance fix for
 CSI-2 node

Message for Open Source:
With this fix we pass v4l2-compliance on "Intel IPU4
CSI-2 0 capture" video node.

Message for Internal:
[Issue/Feature] Run "v4l2-compliance -d /dev/video0" returns:
"fail: !colorspace"
"fail: testColorspace(pix_mp.pixelformat,"
                      pix_mp.colorspace,
                      pix_mp.ycbcr_enc,
                      pix_mp.quantization)
"test VIDIOC_G_FMT: FAIL"

"fail: colorspace >= 0xff"
"test VIDIOC_TRY_FMT: FAIL"

"fail: tot_bytesperline && tot_bytesperline < pix_mp.width"
"fail: Video Capture Multiplanar is valid, but TRY_FMT"
       failed to return a format
"test VIDIOC_TRY_FMT: FAIL"
[Root Cause/Changes] Missing implementation.

Change-Id: I8533b9587d685f53e877f8dc18a7e12769c597da
Tracked-On: #JICCCN-87
Signed-off-by: Yong Zhi <yong.zhi@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/pci/intel/ipu-isys-video.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/media/pci/intel/ipu-isys-video.c b/drivers/media/pci/intel/ipu-isys-video.c
index 0debbe3..c709e4a 100644
--- a/drivers/media/pci/intel/ipu-isys-video.c
+++ b/drivers/media/pci/intel/ipu-isys-video.c
@@ -466,6 +466,11 @@ const struct ipu_isys_pixelformat *ipu_isys_video_try_fmt_vid_mplane(
 	mpix->pixelformat = pfmt->pixelformat;
 	mpix->num_planes = 1;
 
+	mpix->width = clamp(mpix->width, IPU_ISYS_MIN_WIDTH,
+			    IPU_ISYS_MAX_WIDTH);
+	mpix->height = clamp(mpix->height, IPU_ISYS_MIN_HEIGHT,
+			     IPU_ISYS_MAX_HEIGHT);
+
 	if (!av->packed)
 		mpix->plane_fmt[0].bytesperline =
 		    mpix->width * DIV_ROUND_UP(pfmt->bpp_planar ?
@@ -503,8 +508,16 @@ const struct ipu_isys_pixelformat *ipu_isys_video_try_fmt_vid_mplane(
 		    max(mpix->plane_fmt[0].bytesperline,
 			av->isys->pdata->ipdata->isys_dma_overshoot)), 1U);
 
+	memset(mpix->plane_fmt[0].reserved, 0,
+	       sizeof(mpix->plane_fmt[0].reserved));
+
 	if (mpix->field == V4L2_FIELD_ANY)
 		mpix->field = V4L2_FIELD_NONE;
+	/* Use defaults */
+	mpix->colorspace = V4L2_COLORSPACE_RAW;
+	mpix->ycbcr_enc = V4L2_YCBCR_ENC_DEFAULT;
+	mpix->quantization = V4L2_QUANTIZATION_DEFAULT;
+	mpix->xfer_func = V4L2_XFER_FUNC_DEFAULT;
 
 	return pfmt;
 }
-- 
1.9.1

